src/main/service/subscriptionService.ts

```diff
@@ -5,28 +5,44 @@ import { HearingService } from './hearingService';
 import { Hearing } from '../models/hearing';
 import { CourtService } from './courtService';
 import { Court } from '../models/court';
+import {UserSubscriptions} from '../models/UserSubscriptions';
 
 const subscriptionRequests = new SubscriptionRequests();
 const pendingSubscriptionsFromCache = new PendingSubscriptionsFromCache();
 const hearingService = new HearingService();
 const courtService = new CourtService();
 
 export class SubscriptionService {
-  generateCaseTableRows(userid: number): any[] {
-    const subscriptionData = subscriptionRequests.getUserSubscriptions(userid);
+
+  async getSubscriptionsByUser(userid: number): Promise<UserSubscriptions> {
+    const subscriptionData = await subscriptionRequests.getUserSubscriptions(userid);
+    if (subscriptionData) {
+      return subscriptionData;
+    }
+    else {
+      return new class implements UserSubscriptions {
```

Comment:
I am not sure what is this new class implements logic.
Can we just do this please?
`return (subscriptionData) ? subscription data : {caseSubscriptions: [], courtSubscriptions: []};`
Of if you don't like ternary operator
```
if (subscriptionData) {
  return subscriptionData;
} else {
  return {caseSubscriptions: [], courtSubscriptions: []};
}
```

Reply:
changed